src/views/Planning/components/CreateDeliverablePrompt.tsx

```diff
@@ -0,0 +1,65 @@
+import { type MouseEvent } from 'react'
+import { Prompt } from '@/components'
+import { useCollaboration } from '@/hooks/useCollaboration'
+import {
+  article as articleTpl,
+  flash as flashTpl,
+  type TemplatePayload
+} from '@/defaults/templates/'
+import { useSession } from 'next-auth/react'
+import { useRegistry } from '@/hooks/useRegistry'
+
+/**
+ * Deliverable document creation dialog, responsible for creating articles and flashes in the repository.
+ */
+export function CreateDeliverablePrompt({ deliverableType, payload, onClose, title, documentLabel }: {
+  deliverableType: string
+  payload: TemplatePayload
+  title: string
+  documentLabel: string
+  onClose: (
+    event: MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement> | KeyboardEvent,
+    id?: string
+  ) => void
+}): JSX.Element {
+  const { repository } = useRegistry()
+  const { provider } = useCollaboration()
+  const { data: session } = useSession()
+
+  if (!provider?.document || !session?.accessToken || !repository) {
+    return <></>
+  }
+
+  const onCreateDocument = async () => {
+    const id = crypto.randomUUID()
+    await repository.saveDocument(
+      (deliverableType === 'flash') ? flashTpl(id, payload) : articleTpl(id, payload),
+      session.accessToken,
+      BigInt(-1)
+    )
```

Comment:
Might want to error handle this?
```suggestion
    try {
      await repository.saveDocument(
        (deliverableType === 'flash') ? flashTpl(id, payload) : articleTpl(id, payload),
        session.accessToken,
        BigInt(-1)
      )
    } catch (ex) {
      console.error('...')
      toast.error('...')
    }

```

Reply:
The callback itself this is in does not need it's own error handling. The caller is handling the error and displays a error toast.